src/table/body-cell/styles.scss

```diff
@@ -246,9 +242,6 @@ $cell-negative-space-vertical: 2px;
       transition-duration: awsui.$motion-duration-transition-show-quick;
       transition-timing-function: awsui.$motion-easing-sticky;
     }
-    &-pad-left:not(.has-selection):not(.is-visual-refresh.body-cell:first-child.has-striped-rows) {
-      @include cell-padding-inline-start(awsui.$space-table-horizontal);
```

Comment:
Removed this completely. The table cells should use normal paddings also when sticky.